### PR TITLE
Make global scope suggestions consistent with rule scope suggestions.

### PR DESCRIPTION
**Pull request by MikeStall**
This support was recently added (#460)  and only applies to the IGlobalSymbolNameResolver interface, so it's not used by PowerFx.

I hit this as part of #503 , which switches between rule scope and name resolver. These differences were caught by the TestSuggestOptionSets tests.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs [hunk 1]
```diff
@@ -715,11 +715,15 @@ internal static void AddSuggestionsForGlobals(IntellisenseData.IntellisenseData
...
 
-            if (intellisenseData.Binding.NameResolver is IGlobalSymbolNameResolver nr2 && nr2.GlobalSymbols != null)
+            if (intellisenseData.Binding.NameResolver is IGlobalSymbolNameResolver nr2)
             {
-                foreach (var symbol in nr2.GlobalSymbols.Where(symbol => IsMatch(symbol.Key, intellisenseData.MatchingStr)))
+                var globalSymbols = nr2.GlobalSymbols;
+                if (globalSymbols != null)
                 {
-                    CheckAndAddSuggestion(new IntellisenseSuggestion(new UIString(symbol.Key), SuggestionKind.Global, SuggestionIconKind.Other, symbol.Value.Type, -1, symbol.Value.DisplayName, null, null), intellisenseData.Suggestions);
+                    foreach (var symbol in globalSymbols)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseHelper.cs:723]: This is the key part of the fix. 
IsMatch - used here, does a strict prefix match. 
Whereas RuleScope uses AddSuggestionHelper.AddSuggestion, which is a more general "contains" match. 

Make this behave the same was as rule scope.

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by LucGenetier — APPROVED** [review]
:shipit:

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**